.github/workflows/ci.yml

```
@@ -0,0 +1,54 @@
+name: ci
+on:
+  push:
+    branches:
+      - main
```

I wonder if we want this to get triggered on PR, or even any push not only main?  looks like tests run pretty quick in this repo, could be nice to have on every commit.

I think this would accomplish that, right?

```suggestion
on:
  push:
```

---

Yeah absolutely! We can also explicitly trigger on a PR via:

> on:
  pull_request:
    types:
      - opened

Up to y'all really! If you want builds on every push or just every PR 😄

---

We could also add a `workflow_dispatch` so you can dispatch it via the UI at your will on whatever branch you choose to in addition to these.